aten/src/ATen/cuda/CUDASparse.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <ATen/cuda/CUDAContext.h>
+
+// cuSparse Generic API added in CUDA 10.1
+// Windows support added in CUDA 11.0
+// ROCm is not enabled
+#if defined(CUDART_VERSION) && defined(CUSPARSE_VERSION) && ((CUSPARSE_VERSION >= 10300) || (CUSPARSE_VERSION >= 11000 && defined(_WIN32)))
```

Comment:
In the case it's not windows the CUSPARSE_VERSION here is requried to be greater than 10.3, but the [test check below](https://github.com/pytorch/pytorch/pull/60838/files#diff-fe348e24069d43bc7c6913174b038fcc5880a3281bdc0e8e217cf210bd0935e5R170-R175) tests for 10.1, if I'm not miss-understanding the conventions used by CUSPARSE_VERSION

Reply:
CUDA Toolkit uses different version numbers from its components like cuBLAS, cuSOLVER, cuSPARSE. It's actually possible to have a CUDA Toolkit (compilers, etc.) be of version for example 11.4 and cuSPARSE from toolkit of 11.2 (though it'd be a rare situation)
See Table 1 here https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-major-component-versions
In tests, we have no way currently to distinguish between cuda toolkit versions and cusparse, it's also not possible to determine the "update/patch" version (like in `11.1.patch_version`).